lib/perl/Genome/SoftwareResult/User.pm

```diff
@@ -96,16 +99,88 @@ sub _register_users {
     my $label = $newly_created ? 'created' : 'shortcut';
     $user_hash{$label} = $requestor;
 
+    my @all_params;
     while(my ($label, $object) = each %user_hash) {
         my %params = (
             label           => $label,
             user            => $object,
             software_result => $software_result,
         );
-       $class->get(%params) || $class->create(%params);
+        push @all_params, \%params;
+    }
+
+    my $observer;
+    $observer = UR::Context->process->add_observer(
+        aspect => 'precommit',
+        callback => sub {
+            if($observer) {
+                $observer->delete;
+                $observer = undef;
+            } else {
+                Carp::confess 'observer triggered multiple times!';
```

Comment:
The issue we had before is that, if `commit` is triggered while inside the callback, the remaining ones would be queued multiple times even though they are deleted after the first time through.  We no longer call `commit` inside the callback so that isn't an issue anymore.  This is now the equivalent of a `die('This should never happen!');`.

I'll take it out if it seems extraneous.